pkg/s3store/s3store.go

```diff
@@ -373,7 +430,16 @@ func (upload s3Upload) WriteChunk(ctx context.Context, offset int64, src io.Read
 		offset += n
 		bytesUploaded += n
 		nextPartNum += 1
+
+		if err := os.Remove(file.Name()); err != nil {
```

Comment:
If the function returns before os.Remove or file.Close is invoked, we leak file descriptions and temporary files. We also cannot use defer here, since we want to remove files as soon as they are not needed anymore. Do you have an idea how to improve this? Maybe refactor the uploading part into it's own function and use defer in there?

Reply:
Ah, yes, I meant to ask for feedback about this specifically because it's not safe as-is for the reason you mentioned. I think your refactor suggestion is a good one. I'll give that a try and report back. If we're not happy with it, I'm fine with reverting to the current `defer` approach for now.